File: Number Theory/leetcode_866.cpp

// Problem link ===>> https://leetcode.com/problems/prime-palindrome/description/?envType=problem-list-v2&envId=21r1b7y6

#include <bits/stdc++.h>
#include <ext/pb_ds/assoc_container.hpp>
#include <ext/pb_ds/tree_policy.hpp>
using namespace std;
using namespace __gnu_pbds;
#define ll long long
#define ld long double
#define endl '\n'
const int MOD = 1e9 + 7;
const double EPS = 1e-9;
const double PI = acos(-1.0);
template <typename T>
using ordered_set = tree<T, null_type, less<T>, rb_tree_tag, tree_order_statistics_node_update>;
template <typename K, typename V>
using ordered_map = tree<
    K, V, less<K>, rb_tree_tag, tree_order_statistics_node_update>;

class Solution
{
public:
    int primePalindrome(int n)
    {
        if(8 <= n && n<= 11) return 11;
        for (ll x = 1; x < 100000; ++x){
            string s = to_string(x);
            string r(s.rbegin(), s.rend());
            ll y = stoi(s + r.substr(1));
            if (y >= n && isPrime(y)) {
                return y;
            }
        }
        return -1;
    }
    bool isPrime(ll n)
    {
        if (n <2 || n%2 ==0){
            return n == 2;
        }
        for (ll i = 3; i * i <= n; i += 2) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);
    
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        
    }
    return 0;
}

File: Number Theory/leetcode_866.py

# Problem link ===>> https://leetcode.com/problems/prime-palindrome/description/?envType=problem-list-v2&envId=21r1b7y6
# submission link == =>> https://leetcode.com/problems/prime-palindrome/submissions/1723264689/?envType=problem-list-v2&envId=21r1b7y6

import sys
import math
import bisect
import heapq
from collections import defaultdict, deque, OrderedDict, Counter
from functools import lru_cache, reduce
from itertools import permutations, combinations, accumulate
from typing import List, Tuple, Dict, Set, Optional, Union, Any

MOD = 10**9 + 7
INF = float('inf')
input = sys.stdin.read
print = sys.stdout.write


class Solution:
    def primePalindrome(self, n: int) -> int:
        def is_prime(x: int) -> bool:
            if x < 2 or x % 2 == 0:
                return x == 2
            for i in range(3, int(math.isqrt(x)) + 1, 2):
                if x % i == 0:
                    return False
            return True

        if 8 <= n <= 11:
            return 11

        for x in range(1, 100000):
            s = str(x)
            pal = int(s + s[-2::-1])
            if pal >= n and is_prime(pal):
                return pal

        return -1

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())

if __name__ == "__main__":
    solution = Solution()
